### ISSUE TITLE
[python-mlflow] Error collecting package metadata, NotWritableError on /usr/pkgs/urls.txt

### ISSUE THREAD

**Issue by Hydroo24** [acxz/pkgbuilds#24]
I get this error, I don't know if I should give write permissions mlflow ...
I use python-conda AUR package for conda.
```
2020/04/09 09:39:04 INFO mlflow.projects: === Creating conda environment mlflow-266307e4c5272b86e4ea7b8d86fed73b47d05dd6 ===
Warning: you have pip-installed dependencies in your environment file, but you do not list pip itself as one of your conda dependencies.  Conda may not use the correct pip to install your packages, and they may end up in the wrong place.  Please add an explicit pip dependency.  I'm adding one for you, but still nagging you.
Collecting package metadata (repodata.json): failed

NotWritableError: The current user does not have write permissions to a required path.
  path: /usr/pkgs/urls.txt
  uid: 1000
  gid: 1001

If you feel that permissions on this path are set incorrectly, you can manually
change them by executing

  $ sudo chown 1000:1001 /usr/pkgs/urls.txt

In general, it's not advisable to use 'sudo conda'.
```
To reproduce:
```
git clone https://github.com/mlflow/mlflow
cd mlflow/examples
mlflow run sklearn_elasticnet_wine -P alpha=0.5
```

**Comment by acxz** [comment]
hmm, it seems like mlflow uses conda internally? I don't have much ideas, i think we have to go upstream for this one as well.

**Comment by acxz** [comment]
I receive a different error:
```
acxz@archard ~/vcs/git/github/mlflow/mlflow/examples (git)-[master] % mlflow run sklearn_elasticnet_wine -P alpha=0.5         
Traceback (most recent call last):
  File "/usr/bin/mlflow", line 8, in <module>
    sys.exit(cli())
  File "/usr/lib/python3.10/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/usr/lib/python3.10/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/usr/lib/python3.10/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/lib/python3.10/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/lib/python3.10/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/usr/lib/python3.10/site-packages/mlflow/cli.py", line 201, in run
    projects.run(
  File "/usr/lib/python3.10/site-packages/mlflow/projects/__init__.py", line 329, in run
    submitted_run_obj = _run(
  File "/usr/lib/python3.10/site-packages/mlflow/projects/__init__.py", line 102, in _run
    submitted_run = backend.run(
  File "/usr/lib/python3.10/site-packages/mlflow/projects/backend/local.py", line 145, in run
    python_bin_path = _install_python(python_env.python)
  File "/usr/lib/python3.10/site-packages/mlflow/utils/virtualenv.py", line 124, in _install_python
    else _find_latest_installable_python_version(version)
  File "/usr/lib/python3.10/site-packages/mlflow/utils/virtualenv.py", line 100, in _find_latest_installable_python_version
    lines = _exec_cmd(
  File "/usr/lib/python3.10/site-packages/mlflow/utils/process.py", line 95, in _exec_cmd
    process = subprocess.Popen(
  File "/usr/lib/python3.10/subprocess.py", line 971, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/usr/lib/python3.10/subprocess.py", line 1847, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'pyenv'
```

**Comment by acxz** [comment]
I then get the following successful output:
```
acxz@archard ~/vcs/git/github/mlflow/mlflow/examples (git)-[master] % mlflow run sklearn_elasticnet_wine -P alpha=0.5
2023/03/06 12:18:06 INFO mlflow.utils.virtualenv: Installing python 3.8.16 if it does not exist
Downloading Python-3.8.16.tar.xz...
-> https://www.python.org/ftp/python/3.8.16/Python-3.8.16.tar.xz
Installing Python-3.8.16...
Installed Python-3.8.16 to /home/acxz/.pyenv/versions/3.8.16
2023/03/06 12:20:18 INFO mlflow.utils.virtualenv: Creating a new environment /home/acxz/.mlflow/envs/mlflow-eee90753fcbd811c61a366651b95e9611177504d
created virtual environment CPython3.8.16.final.0-64 in 648ms
  creator CPython3Posix(dest=/home/acxz/.mlflow/envs/mlflow-eee90753fcbd811c61a366651b95e9611177504d, clear=False, no_vcs_ignore=False, global=False)
  seeder FromAppData(download=False, pip=bundle, setuptools=bundle, wheel=bundle, via=copy, app_data_dir=/home/acxz/.local/share/virtualenv)
    added seed packages: pip==22.3.1, setuptools==65.6.3, wheel==0.38.4
  activators BashActivator,CShellActivator,FishActivator,NushellActivator,PowerShellActivator,PythonActivator
2023/03/06 12:20:19 INFO mlflow.utils.virtualenv: Installing dependencies

[notice] A new release of pip available: 22.3.1 -> 23.0.1
[notice] To update, run: pip install --upgrade pip

[notice] A new release of pip available: 22.3.1 -> 23.0.1
[notice] To update, run: pip install --upgrade pip
2023/03/06 12:21:29 INFO mlflow.projects.utils: === Created directory /tmp/tmpv6b3mmkn for downloading remote URIs passed to arguments of type 'path' ===
2023/03/06 12:21:29 INFO mlflow.projects.backend.local: === Running command 'source /home/acxz/.mlflow/envs/mlflow-eee90753fcbd811c61a366651b95e9611177504d/bin/activate && python train.py 0.5 0.1' in run with ID '5f69d6de800b47e780f11e3bb610f3f9' === 
Elasticnet model (alpha=0.500000, l1_ratio=0.100000):
  RMSE: 0.7460550348172179
  MAE: 0.576381895873763
  R2: 0.21136606570632266
2023/03/06 12:21:34 INFO mlflow.projects: === Run (ID '5f69d6de800b47e780f11e3bb610f3f9') succeeded ===
mlflow run sklearn_elasticnet_wine -P alpha=0.5  623.52s user 48.93s system 320% cpu 3:29.77 total
```

Looks like this problem has disappeared in the latest `python-mlflow`. As such closing this issue.
